kolibri/plugins/user_profile/assets/test/views/ChangeFacility/ConfirmMerge.spec.js

```diff
@@ -0,0 +1,54 @@
+import { mount, createLocalVue } from '@vue/test-utils';
+import ConfirmMerge from '../../../src/views/ChangeFacility/ConfirmMerge';
+
+const localVue = createLocalVue();
+const sendMachineEvent = jest.fn();
+
+function makeWrapper() {
+  return mount(ConfirmMerge, {
+    provide: {
+      changeFacilityService: {
+        send: sendMachineEvent,
+      },
+    },
+    localVue,
+  });
+}
+
+const getBackButton = wrapper => wrapper.find('[data-test="backButton"]');
+const getContinueButton = wrapper => wrapper.find('[data-test="continueButton"]');
+const clickBackButton = wrapper => getBackButton(wrapper).trigger('click');
+const clickContinueButton = wrapper => getContinueButton(wrapper).trigger('click');
+
+describe(`ChangeFacility/ConfirmMerge`, () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it(`smoke test`, () => {
+    const wrapper = makeWrapper();
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  it('continue is disabled until merging is accepted', async () => {
+    const wrapper = makeWrapper();
+    const checkbox = wrapper.find('input[class="k-checkbox-input"]');
+    const continueButton = wrapper.find('[data-test="continueButton"]');
+    expect(continueButton.vm.disabled).toBeTruthy();
+    checkbox.trigger('click');
+    await wrapper.vm.$nextTick();
+    expect(continueButton.vm.disabled).toBeFalsy();
+    clickContinueButton(wrapper);
+    expect(sendMachineEvent).toHaveBeenCalledWith({
```

Comment:
Nothing blocking, but this part of test actually doesn't test anymore that _'continue is disabled until merging is accepted'_, it rather tests that when the continue button is already enabled, the event will be sent. That's why I think it'd make sense to put this into a separate test case similar to the last test below.

Reply:
yeah, maybe I thought the same but at the end didn't do it because I thought it was going to be too much repeated code. Happy to separate it in a different test.

Reply:
One way to deal with repetition is to group related tests under `describe` and use `beforeEach/beforeaAll` in that `describe` block that would set all tests up. But I don't mind if the test setup is repeated two times or so.